#!/usr/bin/env python
PACKAGE = "rgbd_camera_calibration"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# defaults from https://github.com/Itseez/opencv/blob/master/modules/features2d/src/blobdetector.cpp#L62-L90

gen.add("threshold_step", double_t, 0, "Threshold step", 10.0, 1.0, 100.0)
gen.add("min_threshold", double_t, 0, "Minimum threshold", 50.0, 0.0, 255.0)
gen.add("max_threshold", double_t, 0, "Maximum threshold", 220.0, 0.0, 255.0)

gen.add("min_repeatability", int_t, 0, "Minimum repeatability", 2, 0, 100)

gen.add("min_dist_between_blobs", double_t, 0, "Minimum pixel distance between blobs", 10.0, 0.0, 1000.0)

gen.add("filter_by_color", bool_t, 0, "Filter by color", True)
gen.add("blob_color", int_t, 0, "Blob color", 0, 0, 255)

gen.add("filter_by_area", bool_t, 0, "Filter by area", True)
gen.add("min_area", double_t, 0, "Minimum area", 25.0, 0.0, 10000.0)
gen.add("max_area", double_t, 0, "Maximum area", 5000.0, 0.0, 10000.0)

gen.add("filter_by_circularity", bool_t, 0, "Filter by circularity", False)
gen.add("min_circularity", double_t, 0, "Minimum circularity", 0.8, 0.0, 1.0)
gen.add("max_circularity", double_t, 0, "Maximum circularity", 1.0, 0.0, 1.0)

gen.add("filter_by_inertia", bool_t, 0, "Filter by inertia", True)
gen.add("min_inertia_ratio", double_t, 0, "Minimum inertia ratio", 0.1, 0.0, 1.0)
gen.add("max_inertia_ratio", double_t, 0, "Maximum inertia ratio", 1.0, 0.0, 1.0)

gen.add("filter_by_convexity", bool_t, 0, "Filter by convexity", True)
gen.add("min_convexity", double_t, 0, "Minimum convexity", 0.95, 0.0, 1.0)
gen.add("max_convexity", double_t, 0, "Maximum convexity", 1.0, 0.0, 1.0)

exit(gen.generate(PACKAGE, "rgdb_camera_calibration", "BlobDetector"))
